engine/src/main/java/org/hibernate/validator/internal/engine/constraintvalidation/ConstraintValidatorManager.java

```diff
@@ -327,6 +368,9 @@ private int createHashCode() {
 			int result = annotationDescriptor.hashCode();
 			result = 31 * result + validatedType.hashCode();
 			result = 31 * result + constraintValidatorFactory.hashCode();
+			result = 31 * result + scriptEvaluatorFactory.hashCode();
+			result = 31 * result + clockProvider.hashCode();
+			result = 31 * result + temporalValidationTolerance.hashCode();
```

Comment:
So the idea here would be to have `equals()` and `hashCode()` defined for the initialization context. Thus you wouldn't have to expose anything and it would be better isolated.

The hashCode in the context should be created in the constructor and cached (we have a `createHashCode()` pattern you can see in several places of the code).

Reply:
I guess we'd probably better also use `System#identityHashCode()` and `==` equality as well for `clockProvider` and `scriptEvaluatorFactory` as they can be provided by the users and we have no idea on their impl.